src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java

```diff
@@ -508,24 +512,30 @@ public static void printRootCauseStackTrace(final Throwable throwable, final Pri
      * @return an array of stack trace frames, never null
      * @since 2.0
      */
+    @SuppressWarnings({"index","assignment.type.incompatible"}) /*
+        final Throwable @MinLen(1) [] throwables = getThrowables(throwable); // throwable != null => @MinLen(1)
+        nextTrace = getStackFrameList(throwables[i - 1]); // i != -1 => i - 1 is a valid index for throwables
+        frames.add(throwables[i].toString()); // i is a valid index for throwables
+        frames.add(WRAPPED_MARKER + throwables[i].toString()); // i is a valid index for throwables
+    */
     public static String[] getRootCauseStackTrace(final Throwable throwable) {
         if (throwable == null) {
             return ArrayUtils.EMPTY_STRING_ARRAY;
         }
-        final Throwable throwables[] = getThrowables(throwable);
-        final int count = throwables.length;
+        final Throwable @MinLen(1) [] throwables = getThrowables(throwable); // throwable != null => @MinLen(1)
+        final @Positive int count = throwables.length;
```

Comment:
This annotation should be inferred from the `@MinLen(1)` annotation on the previous line.